dftimewolf/lib/processors/gcp_cloud_resource_tree.py

```diff
@@ -0,0 +1,1034 @@
+      response = request.execute()
+
+      if response:
+        for image in response.get('items', {}):
+          resource = gcp_crt_helper.Resource()
+          resource.id = image.get('id')
+          resource.resource_name = image.get('selfLink')
+          resource.creation_timestamp = image.get('creationTimestamp')
+          if image.get('sourceDisk'):
+            resource.parent = gcp_crt_helper.Resource()
+            resource.parent.resource_name = image.get('sourceDisk')
+            resource.parent.id = image.get('sourceDiskId')
+          elif image.get('sourceSnapshot'):
+            resource.parent = gcp_crt_helper.Resource()
+            resource.parent.resource_name = image.get('sourceSnapshot')
+            resource.parent.id = image.get('sourceSnapshotId')
+
+          result[resource.id] = resource
+
+      # pylint: disable=no-member
+      request = compute_api_client.images().list_next(
+          previous_request=request, previous_response=response)
+
+    return result
+
+  def _RetrieveListOfSnapshots(
+      self, project_id: str) -> Dict[str, gcp_crt_helper.Resource]:
+    """Retrieves list of snapshots in a project.
+
+    Args:
+      project_id: Project Id to retrieve the list of snapshots for
+
+    Returns:
+      Dict of snapshots
+    """
+    result: Dict[str, gcp_crt_helper.Resource] = {}
+
+    # Using beta version of the API because v1 did not have important
+    # information when creating this script
+    compute_api_client = gcp_common.CreateService('compute', 'beta')
+
+    request = compute_api_client.snapshots().list(project=project_id)  # pylint: disable=no-member
+
+    while request is not None:
+      response = request.execute()
+
+      if response:
+        for snapshot in response.get('items', {}):
+          resource = gcp_crt_helper.Resource()
+          resource.id = snapshot.get('id')
+          resource.resource_name = snapshot.get('selfLink')
+          resource.creation_timestamp = snapshot.get('creationTimestamp')
+          if snapshot.get('sourceDisk'):
+            resource.parent = gcp_crt_helper.Resource()
+            resource.parent.resource_name = snapshot.get('sourceDisk')
+            resource.parent.id = snapshot.get('sourceDiskId')
+
+          result[resource.id] = resource
+
+      # pylint: disable=no-member
+      request = compute_api_client.snapshots().list_next(
+          previous_request=request, previous_response=response)
+
+    return result
+
+  def _RetrieveListOfInstances(
+      self, project_id: str) -> Dict[str, gcp_crt_helper.Resource]:
+    """Retrieves list of instances in a project.
+
+    Args:
+      project_id: Project Id to retrieve the list of instances for
+
+    Returns:
+      Dict of instances
+    """
+    result: Dict[str, gcp_crt_helper.Resource] = {}
+
+    # Using beta version of the API because v1 did not have important
+    # information when creating this script
+    compute_api_client = gcp_common.CreateService('compute', 'beta')
```

Comment:
use ListInstances in lcf, it will give you instances keyed by ID after https://github.com/google/cloud-forensics-utils/pull/440

Reply:
I answered this comment before